File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness); 
        int cnt = 0;
        long res=0;
        for(int i=happiness.length-1; i>=happiness.length-k; i--){
             if(happiness[i]+cnt>0){
                res+=(long)happiness[i]+cnt;
             }
             cnt--;
        }
        return res;
    }
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        cnt=0
        res=0
        happiness.sort()
        n = len(happiness)
        for i in range(n-1, n-k-1, -1):
            if happiness[i]+cnt>0:
                res+=happiness[i]+cnt
            cnt-=1
        return res